Ladybird/AppKit/main.mm

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Browser/CookieJar.h>
+#include <Browser/Database.h>
+#include <Ladybird/Utilities.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/EventLoop.h>
+#include <LibGfx/Font/FontDatabase.h>
+#include <LibMain/Main.h>
+
+#import <Application/ApplicationDelegate.h>
+#import <Application/EventLoopImplementation.h>
+#import <UI/Tab.h>
+#import <UI/TabController.h>
+#import <Utilities/URL.h>
+
+#if !__has_feature(objc_arc)
+#    error "This project requires ARC"
+#endif
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    [NSApplication sharedApplication];
+
+    Core::EventLoopManager::install(*new Ladybird::CFEventLoopManager);
+    Core::EventLoop event_loop;
+
+    platform_init();
+
+    // NOTE: We only instantiate this to ensure that Gfx::FontDatabase has its default queries initialized.
+    Gfx::FontDatabase::set_default_font_query("Katica 10 400 0");
+    Gfx::FontDatabase::set_fixed_width_font_query("Csilla 10 400 0");
+
+    StringView url;
+    StringView webdriver_content_ipc_path;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("The Ladybird web browser");
+    args_parser.add_positional_argument(url, "URL to open", "url", Core::ArgsParser::Required::No);
+    args_parser.add_option(webdriver_content_ipc_path, "Path to WebDriver IPC for WebContent", "webdriver-content-path", 0, "path");
+    args_parser.parse(arguments);
+
+#if 0
+    // FIXME: The CFEventLoopImplementation hangs when nested SQL queries are issued, such as when
```

Comment:
:thonk: maybe My earlier comment about NSEvents and the event loop is relevant to this issue.

Reply:
Unfortunately no difference with that change, this was actually the reason I had tried popping multiple events at one point, but it didn't help.

Reply:
Was looking into this earlier, it seems when we issue this nested query to SQLServer, we get stuck in the socket notifier callback:
```c++
static void socket_notifier(CFSocketRef, CFSocketCallBackType, CFDataRef, void const*, void* info)
{
    auto& notifier = *reinterpret_cast<Core::Notifier*>(info);

    Core::NotifierActivationEvent event(notifier.fd());
    notifier.dispatch_event(event); // <---- stuck here
}
```

We enter `Database::execute_statement` for a `SELECT` query, and wait for it to finish with a promise in that function. The socket wakes up and we receive the SQLServer response over it. In the completion callback, we then issue an `INSERT` query. We again enter `Database::execute_statement`, and wait for it to finish. But we're inside a socket notification at this point, and it seems to not re-fire while we are still in it.